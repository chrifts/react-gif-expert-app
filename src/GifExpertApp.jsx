import page1 from './images/1.jpg';
import page2 from './images/2.jpg';
import page2footer from './images/2-footer.jpg';
import page31 from './images/3-1.jpg';
import page32 from './images/3-2.jpg';
import page3Confirm from './images/3-confirm.jpg';
import { useState, useEffect } from 'react';

export const GifExpertApp = () => {

    const [imagenesCargadas, setImagenesCargadas] = useState(false);

    const imagenes = [
        page1,
        page2,
        page2footer,
        page31,
        page32,
        page3Confirm,
    ]; // Agrega otras imágenes aquí según sea necesario


    useEffect(() => {
        const cargarTodasLasImagenes = () => {
            const promesasCarga = imagenes.map((imagen) => {
                return new Promise((resolve, reject) => {
                    const img = new Image();
                    img.onload = resolve;
                    img.onerror = reject;
                    img.src = imagen;
                });
            });

            Promise.all(promesasCarga)
                .then(() => {
                    // Todas las imágenes se han cargado con éxito
                    setImagenesCargadas(true);
                })
                .catch((error) => {
                    // Al menos una imagen falló al cargar
                    console.error('Error al cargar las imágenes:', error);
                });
        };

        cargarTodasLasImagenes();
    }, []);


    const numeroTelefono = '+5491144360229';
    const mensaje = '¡Hola! Quiero confirmar mi asistencia';
    const urlWhatsapp = `https://wa.me/${numeroTelefono}?text=${encodeURIComponent(mensaje)}`;



    return (
        <>
            {imagenesCargadas ? (
                <div style={{ width: '100%', maxWidth: 700 }} className='mx-auto'>
                    <div style={{ position: 'relative' }}>
                        <img src={page1} alt="" />
                    </div>
                    <div style={{ position: 'relative' }}>
                        <img src={page2} alt="" />
                        <div className='position-relative bg-transparent' style={{
                            bottom: 54
                        }}>
                            <p className='text-right m-0 bg-transparent'>
                                <button style={{
                                    position: 'relative',
                                    width: '45%',
                                    height: '50px',
                                    opacity: 0
                                }} className="btn btn-primary" type="button" data-toggle="collapse" data-target="#collapseExample" aria-expanded="false" aria-controls="collapseExample">
                                    Ver mapa
                                </button>
                            </p>
                        </div>
                    </div>
                    <div className='ndBlock'>
                        <div style={{ position: 'relative' }}>
                            <img src={page2footer} alt="" />
                        </div>
                        <div>
                            <div className="collapse" id="collapseExample">
                                <div className="card card-body">
                                    <iframe src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d205.50529295280765!2d-58.72692891336552!3d-34.50073723820536!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x95bca2a3475c9bdd%3A0x3afb7976fc2ff753!2sOasis%20Cooper%20Eventos!5e0!3m2!1sen!2sar!4v1702679616660!5m2!1sen!2sar" width="600" height="450" style={{ border: 0, width: '100%' }} allowFullScreen="" loading="lazy" referrerPolicy="no-referrer-when-downgrade"></iframe>
                                </div>
                            </div>
                        </div>
                        <div style={{ position: 'relative' }}>
                            <img src={page31} alt="" />
                        </div>
                        <div className='w-100 text-center' style={{}}>
                            <a style={{ opacity: 1, display: 'block' }} className="custombtn" href={urlWhatsapp} target="_blank" rel="noopener noreferrer">
                                <img src={page3Confirm} alt="" />
                            </a>
                        </div>
                        <div style={{ position: 'relative' }}>
                            <img src={page32} alt="" />
                        </div>
                    </div>
                </div>
            ) : (
                // Mostrar un indicador de carga mientras se cargan las imágenes
                <p className='text-center'>Cargando...</p>
            )}
        </>
    )
}
