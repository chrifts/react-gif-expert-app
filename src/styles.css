* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(255, 255, 255);
    color: rgb(0, 0, 0);
    overflow-x: hidden;
}

h1, h2, h3 {
    color: rgb(0, 0, 0)
}

body {
    margin: 0;
}

img {
    width: 100%;
}

.divBtn {
    background-color: transparent;
    position: absolute;
    bottom: 383px;
    
    width: 100%;
    text-align: center;
    opacity: 0.2;
}

.divBtn p {
    background-color: transparent;
    margin: 0;
}
.mapBtn {
    bottom: 80px;
}

.ndBlock {
    bottom: 56px;
    position: relative;
}

.custombtn {
    background-color: transparent;
    border: unset;
    border-radius: 30px;
}

.custombtn:focus {
    background-color: transparent;
    border: unset;
    box-shadow: unset;
    opacity: 0;
}