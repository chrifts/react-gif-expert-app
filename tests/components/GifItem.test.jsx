import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Test of <GifItem />', () => {
    const title = 'ImgTest';
    const url = 'https://test-img.com/img.jpg';

    test('should match with snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test('should have the img with alt and src props', () => {
        render(<GifItem title={title} url={url} />)
        //screen.debug();
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('should exists the title', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy();
    });
})