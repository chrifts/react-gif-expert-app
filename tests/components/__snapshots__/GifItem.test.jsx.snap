// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="ImgTest"
      src="https://test-img.com/img.jpg"
    />
    <p>
      ImgTest
    </p>
  </div>
</div>
`;
